### DISCUSSION TITLE
Some random ideas

### DISCUSSION THREAD

**Discussion by olimorris** [mrjones2014/legendary.nvim#234] [Ideas]
Sometimes some random ideas for `legendary.nvim` pop into my head so thought I'd share them:

- Colors - Now we have `itemgroups` we could have colors/highlight groups to make them stand out more (absolutely pointless)
- Shortcuts - Sometimes I have a list of results and the one I want is a few results below. It would be cool to hit a shortcut e.g. `<Ctrl-5>` to access that result:

![Screen Shot 2022-12-05 at 18 08 33@2x](https://user-images.githubusercontent.com/9512444/205710508-e20fb1af-a99c-40a4-94a3-29a716a304e1.png)

- It would be great for `legendary.nvim` to recommend results to me when I open it, based on how many times I've used certain commands in the past

**Comment by mrjones2014** [comment]
>Colors - Now we have itemgroups we could have colors/highlight groups to make them stand out more (absolutely pointless)
Shortcuts - Sometimes I have a list of results and the one I want is a few results below. It would be cool to hit a shortcut e.g. <Ctrl-5> to access that result:

These two I don't think are really possible via `vim.ui.select()`, though they would be possible if I implemented a custom telescope.nvim picker. I've considered it, and I'm open to it if enough users want it, because I would guess that the vast majority of users are using telescope.nvim. Feel free to open an issue just generally about a custom telescope.nvim picker that we can bolt extra features onto.

>It would be great for legendary.nvim to recommend results to me when I open it, based on how many times I've used certain commands in the past

What you're describing is basically the [frecency algorithm](https://en.wikipedia.org/wiki/Frecency), basically sort score is based both on frequency and recency. I've also considered this, but it is a lot more complex than it sounds. It would essentially involve adding a sqlite database to store frecency data, then sorting by that data. However, it also becomes more complex when you think about, how do I resolve the frecency when you change your config? I'd love to do this though. I'd love it if you could open an issue for this!
